src/Symfony/Component/Mailer/Bridge/MicrosoftGraph/Transport/MicrosoftGraphTransport.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\MicrosoftGraph\Transport;
+
+use GuzzleHttp\Psr7\Stream;
+use GuzzleHttp\Psr7\Utils;
+use Microsoft\Graph\Generated\Models\BodyType;
+use Microsoft\Graph\Generated\Models\EmailAddress;
+use Microsoft\Graph\Generated\Models\FileAttachment;
+use Microsoft\Graph\Generated\Models\ItemBody;
+use Microsoft\Graph\Generated\Models\Message;
+use Microsoft\Graph\Generated\Models\ODataErrors\ODataError;
+use Microsoft\Graph\Generated\Models\Recipient;
+use Microsoft\Graph\Generated\Users\Item\SendMail\SendMailPostRequestBody;
+use Microsoft\Graph\GraphServiceClient;
+use Microsoft\Kiota\Authentication\Oauth\ClientCredentialContext;
+use Symfony\Component\Mailer\Bridge\MicrosoftGraph\Exception\SenderNotFoundException;
+use Symfony\Component\Mailer\Bridge\MicrosoftGraph\Exception\SendMailException;
+use Symfony\Component\Mailer\Envelope;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Mime\Header\ParameterizedHeader;
+use Symfony\Component\Mime\Part\DataPart;
+use Symfony\Component\Mime\RawMessage;
+use Symfony\Contracts\Cache\CacheInterface;
+
+class MicrosoftGraphTransport implements TransportInterface
+{
+    private GraphServiceClient $graphServiceClient;
+
+    public function __construct(
+        private readonly string $nationalCloud,
+        private readonly string $tenantId,
+        private readonly string $clientId,
+        private readonly string $clientSecret,
+        private readonly CacheInterface $cache,
+    ) {
+        $tokenRequestContext = new ClientCredentialContext(
+            $this->tenantId,
+            $this->clientId,
+            $this->clientSecret
```

Comment:
Btw, I'm wondering whether we should really take those scalars here vs injecting directly the actual dependencies (when using the factory, the instantiation would happen there). This would be more consistent with the way other bridges with dependencies are implemented (see the amazon bridge for instance)